main.js

```diff
@@ -0,0 +1,110 @@
+// text uit input field opslaan en doorgeven aan button event:
+let textInput = null
+
+const inputField = document.getElementById('input-field')
+inputField.addEventListener('input', (event) => {
+    updateInput(event)
+})
+
+const button = document.getElementById('search-button')
+button.addEventListener('click', findCountryData)
+
+
+// functie die door input field invullen wordt uitgevoerd:
+function updateInput(event) {
+    textInput = event.target.value;
+}
+
+
+// functie die door button click wordt uitgevoerd:
+async function findCountryData() {
+    try {
+        const response = await axios.get(`https://restcountries.eu/rest/v2/name/${textInput}`)
```

Comment:
Wat ik zelf heel vervelend vind is dat ik heel veel `response.data[0]` moet herhalen. Wij developers zijn heel lui, dus is daarom de object spreader in het leven geroepen 😉 
```
const { name: countryName, population, subregion: sbuAreaName, ...otherProps } = response.data[0];
```

_Je hoeft de props niet perse te aliasen_